model.go

```diff
@@ -1176,6 +1192,10 @@ func newModelFromValid(valid map[string]interface{}, importVersion int) (*model,
 		CloudRegion_:   valid["cloud-region"].(string),
 		StatusHistory_: newStatusHistory(),
 	}
+	if rawType, found := valid["type"]; found {
```

Comment:
In other places we have explicitly depended on the importVersion rather than existence of a value in the valid map.